### PR TITLE
#2409 Load bank details from api

### PR DESCRIPTION
**Pull request by desfero**

### CODE REVIEW

File: app/components/wallet/views/start/Start.tsx [hunk 1]
```diff
@@ -120,7 +120,11 @@ export const WalletStartComponent: React.FunctionComponent<TProps> = ({
 
 export const WalletStart = compose<React.FunctionComponent>(
   onEnterAction({
-    actionCreator: dispatch => dispatch(actions.wallet.loadWalletData()),
+    actionCreator: dispatch => {
+      dispatch(actions.wallet.loadWalletData());
+
```
Comment (Moejoe90) [inline app/components/wallet/views/start/Start.tsx]: is this new line of purpose?

File: app/lib/api/client/IHttpClient.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 import { Dictionary } from "../../../types";
+import { Schema } from "../../yup-ts";
 
 export type HttpMethod = "GET" | "POST" | "PUT" | "PATCH" | "DELETE";
 
 export interface IHttpRequestCommon {
   url: string;
   baseUrl?: string;
-  responseSchema?: Yup.Schema<any>;
+  responseSchema?: Yup.Schema<any> | Schema<any>;
```
Comment (Moejoe90) [inline app/lib/api/client/IHttpClient.ts]: in the future we can think about changing this to a generic and get of `any`

Comment (desfero) [reply app/lib/api/client/IHttpClient.ts]: hm, i'm just thinking whether we can get rid of `YupTs` in future and generate type information from `Yup.Schema` in this way `T extends Yup.Schema<infer R> ? R : never`

File: app/components/wallet/views/start/Start.tsx (hunk 1 above)
Comment (desfero) [reply app/components/wallet/views/start/Start.tsx]: yeah, this is on purpose as we will use in next pr data they are downloaded by this action

File: app/lib/api/client/IHttpClient.ts (hunk 2 above)
Comment (Moejoe90) [reply app/lib/api/client/IHttpClient.ts]: If can be done then perfect

File: app/components/wallet/views/start/Start.tsx (hunk 1 above)
Comment (Moejoe90) [reply app/components/wallet/views/start/Start.tsx]: I meant the empty line between them

Comment (desfero) [reply app/components/wallet/views/start/Start.tsx]: ah, it's just for readability, do you think it's not needed?

File: app/modules/kyc/reducer.ts [hunk 3]
```diff
@@ -54,6 +55,9 @@ export interface IKycState {
 
   // contract claims
   claims: TClaims | undefined;
+
+  // api bank details
+  bankAccount: TBankAccount | undefined;
```
Comment (Moejoe90) [inline app/modules/kyc/reducer.ts]: why not `bankAccount?: TBankAccount;`

File: app/modules/kyc/selectors.ts [hunk 4]
```diff
@@ -117,3 +117,5 @@ export const selectIsAccountFrozen = createSelector(selectClaims, claims => {
 
   return false;
 });
+
+export const selectBankAccount = (state: IAppState) => state.kyc.bankAccount;
```
Comment (Moejoe90) [inline app/modules/kyc/selectors.ts]: can you add the return value here its better for readability

File: app/lib/api/client/IHttpClient.ts (hunk 2 above)
Comment (desfero) [reply app/lib/api/client/IHttpClient.ts]: if you can imagine then it can be done 🙏

File: app/lib/api/KycApi.ts [hunk 5]
```diff
@@ -270,4 +269,23 @@ export class KycApi {
...
+      url: BANK_ACCOUNT_PATH,
+      responseSchema: KycBankAccountSchema,
+      allowedStatusCodes: [404],
+    });
+
+    if (response.statusCode === 404) {
+      throw new BankAccountNotFound();
+    }
+
+    return response;
```
Comment (Moejoe90) [inline app/lib/api/KycApi.ts]: I think it's better to return the body here directly to get rid of 
`result.body`

File: app/modules/kyc/sagas.ts [hunk 6]
```diff
@@ -578,6 +580,35 @@ function* submitBusinessRequest(
...
+  logger,
+  notificationCenter,
+}: TGlobalDependencies): Iterator<any> {
+  try {
+    const result: IHttpResponse<TKycBankAccount> = yield apiKycService.getBankAccount();
+
+    yield put(
+      actions.kyc.setBankAccountDetails({
+        hasBankAccount: true,
+        details: result.body,
```
Comment (Moejoe90) [inline app/modules/kyc/sagas.ts]: mentioned above

File: app/modules/kyc/reducer.ts (hunk 3 above)
Comment (desfero) [reply app/modules/kyc/reducer.ts]: as this works in a different way in typescript. In reducer, i want to force this value to be set explicitly by a developer in initialState (see `kycInitialState`). If we use `?` then typescript will not warn us that property should be set in `kycInitialState`.

File: app/modules/kyc/selectors.ts (hunk 4 above)
Comment (desfero) [reply app/modules/kyc/selectors.ts]: fixed

File: app/lib/api/KycApi.ts (hunk 5 above)
Comment (desfero) [reply app/lib/api/KycApi.ts]: fixed

File: app/modules/kyc/sagas.ts (hunk 6 above)
Comment (desfero) [reply app/modules/kyc/sagas.ts]: fixed

File: app/modules/kyc/reducer.ts (hunk 3 above)
Comment (Moejoe90) [reply app/modules/kyc/reducer.ts]: :+1:

**Pull request merged**